src/ai/backend/agent/kubernetes/intrinsic.py

```diff
@@ -279,8 +332,130 @@ async def extra_info(self) -> Mapping[str, str]:
         return {}
 
     async def gather_node_measures(self, ctx: StatContext) -> Sequence[NodeMeasurement]:
-        # TODO: Create our own k8s metric collector
-        return []
+        async def get_memory_stat(core_api, name: str):
+            try:
+                raw_resp = await core_api.connect_get_node_proxy_with_path(
+                    name=name, path="metrics/resource"
+                )
+            except K8sApiException as e:
+                log.warning("kubernetes api error: {}", e)
+                return None
+            resp = [line for line in raw_resp.split("\n") if "#" not in line and "node_mem" in line]
+            mem_used = resp.pop().split(" ")[1]
+            return mem_used
+
+        async def get_network_stat(core_api, name: str):
+            try:
+                raw_resp = await core_api.connect_get_node_proxy_with_path(
+                    name=name, path="metrics/cadvisor"
+                )
+            except K8sApiException as e:
+                log.warning("kubernetes api error: {}", e)
+                return None
+            net_rx = Decimal(0)
+            net_tx = Decimal(0)
+            q = Decimal(0.000)
+            for line in raw_resp.split("\n"):
+                if "#" in line:
+                    continue
+                if "container_network_receive_bytes" in line:
+                    net_rx += Decimal(line.split(" ")[1]).quantize(q)
+                elif "container_network_transmit_bytes" in line:
+                    net_tx += Decimal(line.split(" ")[1]).quantize(q)
+            return net_rx, net_tx
+
+        await K8sConfig.load_kube_config()
```

Comment:
Same here.
Also I'm concerned about whether it would not corrupt the internal states when loading the configuration concurrently from multiple coroutines.